commit c45d8fc758a7b3bcf3dac1897e0070ae7e484cf2
Author: Steven Whitehouse <swhiteho@redhat.com>
Date:   Mon May 24 14:36:48 2010 +0100

    GFS2: Fix permissions checking for setflags ioctl()
    
    We should be checking for the ownership of the file for which
    flags are being set, rather than just for write access.
    
    [Backported to Debian's 2.6.26 by dann frazier <dannf@debian.org>]
    
    Reported-by: Dan Rosenberg <dan.j.rosenberg@gmail.com>
    Signed-off-by: Steven Whitehouse <swhiteho@redhat.com>

diff --git a/fs/gfs2/ops_file.c b/fs/gfs2/ops_file.c
index 34cc876..4d7b436 100644
--- a/fs/gfs2/ops_file.c
+++ b/fs/gfs2/ops_file.c
@@ -224,6 +224,11 @@ static int do_gfs2_set_flags(struct file *filp, u32 reqflags, u32 mask)
 	if (error)
 		return error;
 
+	error = -EACCESS;
+	if (!is_owner_or_cap(inode))
+		goto out;
+
+	error = 0;
 	flags = ip->i_di.di_flags;
 	new_flags = (flags & ~mask) | (reqflags & mask);
 	if ((new_flags ^ flags) == 0)
